applications/NXstress.nxdl.xml

```diff
@@ -0,0 +1,1115 @@
+
+			        * This is the *Hill* system used by Chemical Abstracts.
+		        </doc>
+	        </field>
+            <field name="temperature" type="NX_FLOAT" units="NX_TEMPERATURE" minOccurs="0" maxOccurs="1">
+		        <doc>Sample temperature. This could be a scanned variable</doc>
+		        <dimensions rank="anyRank">
+		                	<dim index="1" value="n_Temp"/><!-- could be any length -->
+		        </dimensions>
+	        </field>
+            <field name="stress_field" type="NX_FLOAT" units="NX_ANY" minOccurs="0" maxOccurs="1">
+		        <doc>Applied external stress field</doc>
+		        <dimensions>
+			            <dim index="1" value="n_sField"/><!-- could be any length -->
+		        </dimensions>
+                <attribute name="direction">
+        	        <enumeration>
+        		        <item value="x"/>
+        		        <item value="y"/>
+        		        <item value="z"/>
+        	        </enumeration>
+                </attribute>
+	        </field>
+            <group name="gauge_volume" type="NXparameters" minOccurs="0" maxOccurs="1">
+                <doc> The gauge volume can be described with the following parameters:
+                 .. figure:: stress/GV2.jpg
+                  :width: 70%
+                  :alt: Dimensions of the gauge volume.
+
+                  Parameters describing the gauge volume.
+                </doc>
+                <field name="x" type="NX_FLOAT" units="NX_LENGTH" minOccurs="0" maxOccurs="1">
+                    <doc>
+                        Length of the first diagonal.
+                    </doc>
+                </field>
+                <field name="y" type="NX_FLOAT" units="NX_LENGTH" minOccurs="0" maxOccurs="1">
+                    <doc>
+                        Length of the second diagonal normal to :ref:`x &lt;/NXstress/ENTRY/sample_description/gauge_volume/x-field&gt;`.
+                    </doc>
+                </field>
+                <field name="z" type="NX_FLOAT" units="NX_LENGTH" minOccurs="0" maxOccurs="1">
+                    <doc>
+                        Height of the gauge volume.
+                    </doc>
+                </field>            
+                <field name="depends_on" type="NX_CHAR" minOccurs="0" maxOccurs="1">
+                    <doc>
+                    The axis on which the sample position depends may be stored
+                    anywhere, but is normally stored in the NXtransformations
+                    group within the NXsample group.
+                    </doc>
+                </field>
+            </group>
+            <group type="NXtransformations" minOccurs="0" maxOccurs="1">
+                <doc>
+                This is the recommended location for sample goniometer
+                and other related axes.
+                </doc>
+            </group>
+        </group>
+
+        <group name="FIT" nameType="any" type="NXprocess" minOccurs="1" maxOccurs="unbounded">

```

Comment:
The new MPES application definition has `NXfit`, etc. Perhaps this could be migrated to use those new base classes.

Reply:
For reference:

https://github.com/nexusformat/definitions/pull/1424/files#diff-0bd69f0a4c1bf5a303bd1c85a4fbf265bc64416513d6cec9af6212934b1274d8

https://github.com/nexusformat/definitions/pull/1424/files#diff-3baad8533e4f0fa71dcd336841c1ed902c77d00e3f63cdba7b8a83298d59ba6b

That would require work and time and I don't have any.

Reply:
Also it seems like NXfit decided to have one group per peak. That is very inconvenient. Here we decided to have an NXparameters group fit a 1D field per fit parameters.

Reply:
Maybe it will be good to work with the FAIRmat people (@lukaspie) to see if you can make common base classes for fitting - they have multiple per-peak groups whereas you have the multiple peaks per group as you are using the same function (peak/profile type) for each peak.